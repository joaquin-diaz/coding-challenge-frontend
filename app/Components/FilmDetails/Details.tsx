import * as React from 'react';
import Film from '../../types/Film';
import styled from '@emotion/styled';

interface DetailsProps {
  film: Film;
}

const FilmTitle = styled.div`
  font-size: 20px;
  margin: 10px 0;
`;

const Actors = styled.div`
  font-style: italic;
`;

const Details: React.FC<DetailsProps> = props => {
  const { film } = props;

  return (
    <div>
      <FilmTitle>
        {film.title} - {film.releaseYear}
      </FilmTitle>
      <strong>Directed by</strong>: {film.director} <br />
      <Actors>
        {film.actor1}, {film.actor2}, {film.actor3}
      </Actors>
    </div>
  );
};

export default Details;
