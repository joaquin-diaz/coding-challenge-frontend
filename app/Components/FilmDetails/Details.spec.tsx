import * as React from 'react';

import Details from './Details';
import { mount } from 'enzyme';
import { fakeFilm } from '../../utils/test.fixtures';

describe('<Details />', () => {
  it('should render', () => {
    const wrapper = mount(<Details film={fakeFilm} />);
    expect(wrapper).toMatchSnapshot();
  });
});
