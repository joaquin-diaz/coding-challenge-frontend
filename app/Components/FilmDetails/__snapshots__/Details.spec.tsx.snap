// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render 1`] = `
.emotion-0 {
  font-size: 20px;
  margin: 10px 0;
}

.emotion-1 {
  font-style: italic;
}

<Details
  film={
    Object {
      "actor1": "Siddarth",
      "actor2": "Nithya Menon",
      "actor3": "Priya Anand",
      "address": "Epic Roasthouse (399 Embarcadero)",
      "coordinates": Object {
        "lat": 123,
        "lng": 456,
      },
      "director": "Jayendra",
      "productionCompany": "SPI Cinemas",
      "releaseYear": "2011",
      "title": "180",
      "writer": "Umarji Anuradha, Jayendra, Aarthi Sriram, & Suba ",
    }
  }
>
  <div>
    <Styled(div)>
      <div
        className="emotion-0"
      >
        180
         - 
        2011
      </div>
    </Styled(div)>
    <strong>
      Directed by
    </strong>
    : 
    Jayendra
     
    <br />
    <Styled(div)>
      <div
        className="emotion-1"
      >
        Siddarth
        , 
        Nithya Menon
        , 
        Priya Anand
      </div>
    </Styled(div)>
  </div>
</Details>
`;
